### PR TITLE
Functionality in slot booking page, fixed requested bugs, resolved conflicts created by recent PRs 

### PR DESCRIPTION
**Pull request by Ayushsingh2110**

### CODE REVIEW

**Comment by Ayushsingh2110** [comment]
Check commits made after June 30 2024, commits before that are already accepted in previous PRs.

File: frontend/src/components/User/Therapists/Our_Therapist.js [hunk 1]
```diff
@@ -1,203 +0,0 @@
-import React, { useState, useEffect } from "react";
```
Comment (chandan24042001s) [inline frontend/src/components/User/Therapists/Our_Therapist.js:1]: Why U deleted this Component ?

Have U checked the UI.
Are U kidding me ?

File: frontend/src/components/PopUps/LoginReq_pop.js [hunk 2]
```diff
@@ -1,38 +1,41 @@
-import React, { useEffect, useRef } from 'react'
```
Comment (chandan24042001s) [inline frontend/src/components/PopUps/LoginReq_pop.js:1]: Where is the Cross icon on the top ?

Why blindly following the Figma  ? Use Mind. How user is going to know how to close the popup ?

File: frontend/src/components/User/Register/RegisterLogin.js [hunk 3]
```diff
@@ -81,9 +81,8 @@ const Register__Login = () => {
 
       const data = await response.json();
 
-      dispatch(addUser(data.newUser));
+      dispatch(addUser({_id: data.newUser._id }));
```
Comment (chandan24042001s) [inline frontend/src/components/User/Register/RegisterLogin.js:84]: Why only UserId ? We are not fetching anything from Id now ?

File: frontend/src/utils/Hooks/fetchData.js [hunk 4]
```diff
@@ -13,10 +13,11 @@ const useFetch = (url) => {
       try {
         const res = await axios.get(url);
         setData(res.data);
-        setStatus(res.status);
+
+        //setStatus(res.status);
         console.log(res)
```
Comment (chandan24042001s) [inline frontend/src/utils/Hooks/fetchData.js:18]: Forget to remove this console .

File: frontend/src/components/PopUps/LoginReq_pop.js [hunk 5]
```diff
@@ -1,38 +1,41 @@
...
+      <div
+        className="relative w-[70%] bg-white rounded-lg shadow"
+        onClick={handlePopupClick}
+      >
+        <div className="text-center border-b-2 border-[#0190B1]">
+          <h3 className="w-[100%] my-5 text-[28px] md:text-[24px] sm:text-[20px] text-[#0190B1]">
+            Login Required
+          </h3>
+        </div>
+        <div className="my-[24px] mx-[15vw]">
```
Comment (chandan24042001s) [inline frontend/src/components/PopUps/LoginReq_pop.js:27]: Why pixel and Vh ? We are replacing it with % .

File: frontend/src/components/User/Therapists/TherapistDetailsPage.js [hunk 6]
```diff
@@ -68,43 +70,43 @@ function TherapistDetailsPage() {
...
+                  strokeWidth="2"
                   d="m19 19-4-4m0-7A7 7 0 1 1 1 8a7 7 0 0 1 14 0Z"
                 />
               </svg>
             </div>
             <input
               type="search"
               id="default-search"
-              class="block w-[743px] p-4 ps-10 text-sm rounded-lg bg-[#EFF7FF] border border-[#B4F0FF] outline-none"
+              className="block w-[743px] p-4 ps-10 text-sm rounded-lg bg-[#EFF7FF] border border-[#B4F0FF] outline-none"
```
Comment (chandan24042001s) [inline frontend/src/components/User/Therapists/TherapistDetailsPage.js:101]: Why Pixel ? 743px😨. 

We have % for Screen Calculation ?

File: frontend/src/components/PopUps/LoginReq_pop.js [hunk 7]
```diff
@@ -1,38 +1,41 @@
...
 
-const LoginReq_pop = ({ setShowPopUp }) => {
+const LoginReqPop = ({ setShowPopUp }) => {
+  const handleClose = () => {
+    setShowPopUp(false);
+  };
 
-    const PopUpRef = useRef();
+  const handlePopupClick = (e) => {
+    e.stopPropagation(); // Prevent click event from bubbling up to the overlay
```
Comment (chandan24042001s) [inline frontend/src/components/PopUps/LoginReq_pop.js:10]: Are U still manipulating DOM tree ?

File: frontend/src/components/Booking/IntakeResponse.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+    function handleChange(e){
+        const name = e.target.name
+        const value = e.target.value
+        setIntakeForm((obj) => ({...obj, [name]: value}))
+    }
+
+    function handleFormSubmit(e){
+        e.preventDefault();
+        console.log(IntakeForm)
```
Comment (chandan24042001s) [inline frontend/src/components/Booking/IntakeResponse.js:19]: ?

File: frontend/src/components/User/Register/RegisterLogin.js (hunk 3 above)
Comment (Ayushsingh2110) [reply frontend/src/components/User/Register/RegisterLogin.js:84]: > Why only UserId ? We are not fetching anything from Id now ?

We will use the userID stored in localstorage, and will use token to authenticate, I don't see any requirement to store user data in redux, it is better to remove it.

File: frontend/src/components/PopUps/LoginReq_pop.js (hunk 7 above)
Comment (Ayushsingh2110) [reply frontend/src/components/PopUps/LoginReq_pop.js:10]: > Are U still manipulating DOM tree ?

DOM is not getting manipulated here, If you see any possible alternative then tell me

File: frontend/src/components/Booking/IntakeResponse.js (hunk 8 above)
Comment (Ayushsingh2110) [reply frontend/src/components/Booking/IntakeResponse.js:19]: > ?

After selecting slots, the user will then get redirected to intake form, here I'm just checking if form is getting filled rightly or not, further work will be done

File: frontend/src/components/User/Therapists/TherapistDetailsPage.js (hunk 6 above)
Comment (Ayushsingh2110) [reply frontend/src/components/User/Therapists/TherapistDetailsPage.js:101]: > Why Pixel ? 743px😨.
> 
> We have % for Screen Calculation ?

I don't have any responsive UI design with me, responsiveness can tackled after completing work on functionality

File: frontend/src/components/PopUps/LoginReq_pop.js (hunk 5 above)
Comment (Ayushsingh2110) [reply frontend/src/components/PopUps/LoginReq_pop.js:27]: > Why pixel and Vh ? We are replacing it with % .
Replace it after work is completed on functionality, responsiveness can be looked later

File: frontend/src/utils/Hooks/fetchData.js (hunk 4 above)
Comment (Ayushsingh2110) [reply frontend/src/utils/Hooks/fetchData.js]: > Forget to remove this console .

Console is now removed in latest commits

File: frontend/src/components/PopUps/LoginReq_pop.js (hunk 2 above)
Comment (Ayushsingh2110) [reply frontend/src/components/PopUps/LoginReq_pop.js:1]: > Where is the Cross icon on the top ?
> 
> Why blindly following the Figma ? Use Mind. How user is going to know how to close the popup ?

cross is now added in latest commits

File: frontend/src/components/User/Therapists/Our_Therapist.js (hunk 1 above)
Comment (Ayushsingh2110) [reply frontend/src/components/User/Therapists/Our_Therapist.js:1]: > Why U deleted this Component ?
> 
> Have U checked the UI. Are U kidding me ?

Duplicate file was removed, original Our_therapist is now modified with latest changes made in UI, check in recent commits

**Pull request merged**